tests/test_basic_ops_dict.py

```diff
@@ -364,13 +384,19 @@ def test_iterate_dict_with_interpolation() -> None:
 def test_dict_pop(cfg: Dict[Any, Any], key: Any, default_: Any, expected: Any) -> None:
     c = OmegaConf.create(cfg)
 
-    if default_ != "__NO_DEFAULT__":
-        val = c.pop(key, default_)
-    else:
-        val = c.pop(key)
+    def pop() -> Any:
+        if default_ != "__NO_DEFAULT__":
+            return c.pop(key, default_)
+        else:
+            return c.pop(key)
 
-    assert val == expected
-    assert type(val) == type(expected)
+    if isinstance(expected, RaisesContext):
+        with expected:
+            pop()
+    else:
+        val = pop()
+        assert val == expected
+        assert type(val) == type(expected)
```

Comment:
Break into positive and negative tests.
Edit:
reading further, I see you touched test_dict_struct_mode_pop. why not move the error test cases there?

Reply:
So actually I didn't touch that test (it's Github's diff view that's confusing). But it should have been in `test_dict_pop_error()` => done in 1f0601a